### PR TITLE
feat: star routing container pea

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/peapods/peas/factory.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from . import BasePea, Pea
+from .container import ContainerPea
+
+from ...hubble.helper import is_valid_huburi
+from ...hubble.hubio import HubIO
+
+
+class PeaFactory:
+    """
+    A PeaFactory is a factory class, abstracting the PeaFactory creation
```
Comment (jacobowitz) [inline jina/peapods/peas/factory.py:13]:
```suggestion
    A PeaFactory is a factory class, abstracting the Pea creation
```

File: jina/peapods/peas/factory.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    """
+
+    @staticmethod
+    def build_pea(
+        args: 'Namespace',
+    ) -> BasePea:
+        """Build an implementation of a `BasePod` interface
```
Comment (jacobowitz) [inline jina/peapods/peas/factory.py:20]:
```suggestion
        """Build an implementation of a `BasePea` interface
```

File: jina/peapods/runtimes/asyncio.py [hunk 3]
```diff
@@ -125,22 +131,48 @@ def activate(**kwargs):
...
+        Check if status is ready.
+
+        :param ctrl_address: the address where the control message needs to be sent
+        :param kwargs: extra keyword arguments
+
+        :return: True if status is ready else False.
+        """
+
+        try:
+            GrpcConnectionPool.send_message_sync(ControlMessage('STATUS'), ctrl_address)
```
Comment (jacobowitz) [inline jina/peapods/runtimes/asyncio.py:146]: I am not sure that really belongs here? the grpc server is created by the child class implementations.

File: jina/peapods/runtimes/gateway/grpc/__init__.py [hunk 4]
```diff
@@ -40,6 +40,7 @@ async def async_setup(self):
         jina_pb2_grpc.add_JinaRPCServicer_to_server(self.streamer, self.server)
         bind_addr = f'{__default_host__}:{self.args.port_expose}'
         self.server.add_insecure_port(bind_addr)
+        self.logger.debug(f' Start server bound at {bind_addr}')
```
Comment (jacobowitz) [inline jina/peapods/runtimes/gateway/grpc/__init__.py:43]:
```suggestion
        self.logger.debug(f' Start server bound to {bind_addr}')
```

**Review by jacobowitz — CHANGES_REQUESTED** [review]
Some smaller comments, otherwise looks good.
I was just wondering if we should add a test trying to create ContainerPeas from within a container? Or is this a challenge we leave for jinad?

File: jina/peapods/runtimes/asyncio.py (hunk 3 above)
Comment (JoanFM) [reply jina/peapods/runtimes/asyncio.py:146]: But it is the only way ContainerPea can avoid knowing runtime_cls. I agree this should be done as discussed via the container itself, but not possible now

**Comment by JoanFM** [comment]
> Some smaller comments, otherwise looks good. I was just wondering if we should add a test trying to create ContainerPeas from within a container? Or is this a challenge we leave for jinad?

This should be a challenge for JinaD yes I'd say.

**Review by jacobowitz — APPROVED** [review]
:rocket:

**Pull request merged**